src/main/java/com/es/phoneshop/web/CheckoutPageServlet.java

```diff
@@ -0,0 +1,130 @@
+        orderService = DefaultOrderService.getInstance();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Cart cart = cartService.getCart(request);
+        request.setAttribute("order", orderService.getOrder(cart));
+        request.setAttribute("paymentMethods", orderService.getPaymentMethods());
+        request.getRequestDispatcher(CHECKOUT_JSP).forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Cart cart = cartService.getCart(request);
+        Order order = orderService.getOrder(cart);
+
+        Map<String, String> errors = new HashMap<>();
+
+        setRequiredParameter(request, "firstName", errors, order::setFirstName);
+        setRequiredParameter(request, "lastName", errors, order::setLastName);
+        setRequiredParameter(request, "phone", errors, order::setPhone);
+        setRequiredParameter(request, "deliveryAddress", errors, order::setDeliveryAddress);
+        setPaymentMethod(request, errors, order);
+        setDeliveryDate(request, errors, order);
+
+        handleErrors(request, response, errors, order);
+    }
+
+    private LocalDate getDeliveryDate(HttpServletRequest request) {
+        String deliveryDateString = request.getParameter("deliveryDate");
+        int day = Integer.parseInt(deliveryDateString.substring(0, 2));
+        int month = Integer.parseInt(deliveryDateString.substring(3, 5));
+        int year = Integer.parseInt(deliveryDateString.substring(6, 10));
+        return LocalDate.of(year, month, day);
+    }
+
+    private void setRequiredParameter(HttpServletRequest request, String parameter, Map<String, String> errors,
+                                      Consumer<String> consumer) {
+        String value = request.getParameter(parameter);
+        if (value == null || value.isEmpty()) {
+            errors.put(parameter, "Value name is required");
+        } else {
+            consumer.accept(value);
+        }
+    }
+
+    private void setPaymentMethod(HttpServletRequest request, Map<String, String> errors, Order order) {
+        String paymentMethod = request.getParameter("paymentMethod");
+        if (paymentMethod == null || paymentMethod.isEmpty()) {
+            errors.put("paymentMethod", "Value is required");
+        } else {
+            order.setPaymentMethod(PaymentMethod.valueOf(paymentMethod));
+        }
+    }
+
+    private void setDeliveryDate(HttpServletRequest request, Map<String, String> errors, Order order) {
+        String deliveryDate = request.getParameter("deliveryDate");
+        if (deliveryDate == null || deliveryDate.isEmpty()) {
+            errors.put("deliveryDate", "Value is required");
+        } else {
+            try {
+                int day = Integer.parseInt(deliveryDate.substring(0, 2));
```

Comment:
Don't like it, there are more convenient and modern ways to parse date upon some format e.x. SimpleDateFormat class. It's hint, do not change it, but keep in mind it for future, the right way is to use java.util classes introduced in java8 when working with date and time.